### PR TITLE
Add update user endpoint

### PR DESCRIPTION
**Pull request by HubertTatar**
Update user endpoint + list endpoint for convenient way to see data in db.

### CODE REVIEW

**Comment by HubertTatar** [comment]
Need to figure out why tests are failing on Travis.

File: functional_test/live_tests/users_test.py [hunk 1]
```diff
@@ -36,3 +36,25 @@ def test_signup_existing_user(pet_store_client):
...
+        "email": "wheresmycar@gmail.com",
+        "password": "wickofyourwit",
+        "phone": "215-789-0123",
+        "id": 1
+    }
+    response = pet_store_client.update_user(user)
+    assert_that(response.status_code, is_(200))
+
+    updated_user = response.json()
+    assert_that(updated_user['lastName'], is_('Wicked'))
```
Comment (pauljamescleary) [inline functional_test/live_tests/users_test.py]: need a new line at the end here

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala [hunk 2]
```diff
@@ -4,14 +4,21 @@ import cats.effect.Effect
...
+
+import scala.language.higherKinds
+import io.github.pauljamescleary.petstore.domain.{UserAlreadyExistsError, UserNotFoundError}
+import io.github.pauljamescleary.petstore.domain.users.{User, UserService}
 
 class UserEndpoints[F[_]: Effect] extends Http4sDsl[F] {
 
+  import QueryParamDecoder._
+
+  object PageSizeMatcher extends QueryParamDecoderMatcher[Int]("pageSize")
```
Comment (pauljamescleary) [inline src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala]: we are using the same decoders pageSize and offset for pets.  Perhaps pull these into a separate module and import in both?

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobieUserRepositoryInterpreter.scala [hunk 3]
```diff
@@ -45,6 +45,14 @@ class DoobieUserRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
       case None =>
         none[User].pure[F]
     }
+
```
Comment (pauljamescleary) [inline src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobieUserRepositoryInterpreter.scala]: The other functions here do not use `override`, should remove the `override` for consistency

File: functional_test/live_tests/users_test.py (hunk 1 above)
Comment (HubertTatar) [reply functional_test/live_tests/users_test.py]: after assert or before?

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala (hunk 2 above)
Comment (HubertTatar) [reply src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala]: By separate module you mean trait i.e. named 'Pagination' ?

File: functional_test/live_tests/users_test.py (hunk 1 above)
Comment (pauljamescleary) [reply functional_test/live_tests/users_test.py]: all files should have an empty line at the end of the file

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala (hunk 2 above)
Comment (pauljamescleary) [reply src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala]: An object Pagination would be sufficient, and we could import the matchers.

File: functional_test/pet_store_client.py [hunk 4]
```diff
@@ -132,4 +132,21 @@ def signup_user(self, user):
...
+        return self.session.request('PUT', url, self.headers, json.dumps(user))
+
+    def list_users(self, page_size=10, offset=0):
+        """
+        Returns a list of users
+        :return:
+        """
+        url = urljoin(self.index_url, "/users?pageSize={0}&offset={1}".format(page_size, offset))
 
+        return self.session.request('GET', url, self.headers)
```
Comment (pauljamescleary) [inline functional_test/pet_store_client.py]: here we need a new line at the end of the file, otherwise we get these red little arrows

**Pull request merged**